Assets/Scripts/Tracker/BaseTracker.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using System.Collections.Generic;
+using Logger = Mediapipe.Unity.Logger;  // Disambiguation with UnityEngine.Logger
+using Mediapipe;
+using Mediapipe.Unity;
+using Mediapipe.Unity.Holistic;
+using UnityEngine;
+
+public abstract class BaseTracker : MonoBehaviour {
+  private const string _TAG = nameof(BaseTracker);
+  public enum InferenceMode {
+    CPU,
+    GPU,
+  }
+  [Tooltip("Inference mode.")]
+  [SerializeField] private InferenceMode _preferableInferenceMode;
+  [Tooltip("Reference to the internal texture frame pool.")]
+  [SerializeField] private TextureFramePool _textureFramePool;
+  [SerializeField] private Texture _sourceTexture;
+  [Tooltip("Whether to flip the source image horizontally before processing.")]
+  [SerializeField] private bool _hFlip;
+  [Tooltip("Whether to flip the source image vertically before processing.")]
+  [SerializeField] private bool _vFlip;
+  [Tooltip("Angles to rotate the source image.")]
+  [SerializeField] private int _rotation;
+  protected HolisticTrackingGraph _graphRunner;
+  private Coroutine _coroutine;
+  private InferenceMode _inferenceMode;
+
+  // <summary>Attaches any event handler for callbacks. </summary>
+  public abstract void AddEventHandler();
+
+  public IEnumerator Start() {
+    _graphRunner = GetComponent<HolisticTrackingGraph>();
+    AssetLoader.Provide(new StreamingAssetsResourceManager());
+
+    Logger.LogInfo(_TAG, "Starting mediapipe");
+    DecideInferenceMode();
+    if (_inferenceMode == InferenceMode.GPU) {
+      Logger.LogInfo(_TAG, "Initializing GPU resources...");
+      yield return GpuManager.Initialize();
+    }
+    var graphInitRequest = _graphRunner.WaitForInit(RunningMode.Async);
```

Comment:
No particular reason. It is just how the [ImageSourceSolution](https://github.com/homuler/MediaPipeUnityPlugin/blob/6b8c6743f23539f7604e74dc260b01e0f58f1707/Assets/MediaPipeUnity/Samples/Common/Scripts/ImageSourceSolution.cs) works. Moved the resize call to the top.

Unfortunately it will still need the isError call on the returned object so it can't remove the local variable `graphInitRequest`.